### PR TITLE
Feature/qa 15623 censimento stime mittenti refactor

### PR DESCRIPTION
**Pull request by mmancini95**
This pull request introduces several improvements to the management and testing of sender limit estimates, particularly focusing on enhanced logging, more robust test scenarios, and improved data handling for commessa (order module) files. The main changes include the addition of new test data files and scenarios, enhancements to logging for better traceability during test execution, and adjustments to the logic for handling estimates that span across months.

**Test data and scenario enhancements:**

* Added new JSON files (`modulo_commessa_P1_giugno.json`, `modulo_commessa_P1_giugno_modified.json`, `modulo_commessa_P1_luglio.json`) to support advanced test cases involving sender limit estimates across month boundaries. [[1]](diffhunk://#diff-5d33870d68834cf58a5942c8bce6dbee2d621d3d16a3fb844b54fc9d45f517acR1-R63) [[2]](diffhunk://#diff-11e27091d3e6ee52d27748788d98e6b08c08847e70b00965f1d1b3c31f5f5b09R1-R63) [[3]](diffhunk://#diff-8faf18121c289139ed7bf24a653268d6f105f57d8880915a29d8ddf7da6ba53bR1-R63)
* Updated the feature file `CensimentoStimeMittenti.feature` to introduce a new scenario that verifies correct updating of `originalEstimate` and `monthlyEstimate` when a commessa is reloaded for the same month, especially for weeks spanning two months. Removed an outdated scenario and improved test coverage for edge cases. [[1]](diffhunk://#diff-d6c036de5c68c511a3e9ca05469a38e5f2418a225595242263801cfb72d13ea9L9-L27) [[2]](diffhunk://#diff-d6c036de5c68c511a3e9ca05469a38e5f2418a225595242263801cfb72d13ea9R164-R186) [[3]](diffhunk://#diff-d6c036de5c68c511a3e9ca05469a38e5f2418a225595242263801cfb72d13ea9L196) [[4]](diffhunk://#diff-d6c036de5c68c511a3e9ca05469a38e5f2418a225595242263801cfb72d13ea9L213-L218)

**Logging improvements:**

* Enhanced logging by adding new MDC entries for `testCaseId` and `featureFile` in `RestTemplateConfiguration` and extracting these values in `SharedSteps`. This enables more granular and contextual log tracing during test execution. [[1]](diffhunk://#diff-6f3c9e1b53c6458dac23888973bad78a592bf0cea19412aa2acad3a572a6d0a0R33-R34) [[2]](diffhunk://#diff-6f8c0647b33e4c34fd9fd13308612310275da86bc4c72f62eea5dba627df3b1aR89-R90) [[3]](diffhunk://#diff-6f8c0647b33e4c34fd9fd13308612310275da86bc4c72f62eea5dba627df3b1aR279-R292)
* Introduced a custom `logback-spring.xml` configuration to include the new MDC fields in log output and to flatten stack traces for better readability in log aggregators.

**Test robustness and correctness:**

* Increased Awaitility timeouts in test steps to reduce flakiness and improve reliability for operations that may take longer in certain environments. [[1]](diffhunk://#diff-0ae0075ebc43dd1858d2c6d2e199eab7b70aee969860b04c61fd87b763fc7170L66-R66) [[2]](diffhunk://#diff-0ae0075ebc43dd1858d2c6d2e199eab7b70aee969860b04c61fd87b763fc7170L209-R211)
* Enhanced the matching logic in sender limit assertions to include checks for `monthlyEstimate` and `originalEstimate`, ensuring stricter validation of test outcomes.

**Estimate handling logic:**

* Updated `ModuloCommessa` to set both `monthlyEstimate` and `originalEstimate` when generating sender limits, ensuring that these values are correctly populated for downstream processes and tests.

**Other minor changes:**

* Small update to a template engine test to match new expected output text.
* Added missing imports needed for new logic in `SharedSteps`.

These changes collectively improve the reliability, traceability, and correctness of sender limit estimate handling and related automated tests.

### CODE REVIEW

File: src/test/java/it/pagopa/pn/cucumber/steps/SharedSteps.java [hunk 1]
```diff
@@ -271,9 +276,20 @@ public void afterScenario() {
     public void injectScenarioNameInsideSfl4jMdc(Scenario scenario) {
```
[the remark this answers is not in the document]
Comment (mmancini95) [reply src/test/java/it/pagopa/pn/cucumber/steps/SharedSteps.java]: Risolto

File: src/test/resources/logback-spring.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <!--
+        Pattern derivato dal default di Spring Boot con tre differenze:
+        - viene aggiunto [%X{testCaseId:-N/A}], valorizzato in SharedSteps#injectScenarioNameInsideSfl4jMdc
```
[the remark this answers is not in the document]
Comment (mmancini95) [reply src/test/resources/logback-spring.xml]: Corretto

File: src/test/java/it/pagopa/pn/cucumber/steps/censimentoStimeMittenti/CensimentoStimeMittentiSteps.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ public void fetchSenderLimitUntilCondition(String senderLimitTable, boolean shou
     private void assertSenderLimitMatchesEventually(String senderLimitTable, String province, DelayerSenderLimit expectedSenderLimit, boolean shouldContain) {
         try {
             Awaitility.await()
-                    .atMost(Duration.ofMinutes(5))
+                    .atMost(Duration.ofMinutes(20))
                     .pollInterval(Duration.ofSeconds(10))
```
[the remark this answers is not in the document]
Comment (mmancini95) [reply src/test/java/it/pagopa/pn/cucumber/steps/censimentoStimeMittenti/CensimentoStimeMittentiSteps.java:67]: Il tempo è un tempo massimo ed a volte è necessario poiché il caricamento non è immediato.

**Review by tulli086 — APPROVED** [review]
